backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java

```diff
@@ -0,0 +1,91 @@
+package pl.wroclaw.asi.labdaybackendspring.controllers;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pl.wroclaw.asi.labdaybackendspring.model.AppData;
+import pl.wroclaw.asi.labdaybackendspring.model.LastUpdate;
+import pl.wroclaw.asi.labdaybackendspring.services.*;
+
+import java.security.Principal;
+import java.sql.Timestamp;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api")
+public class AppDataController {
+
+    private final EventService eventService;
+    private final PlaceService placeService;
+    private final PathService pathService;
+    private final TimetableService timetableService;
+    private final SpeakerService speakerService;
+    private final LastUpdateService lastUpdateService;
+
+    private final ValidationErrorService validationErrorService;
+
+    public AppDataController(
+            EventService eventService,
+            PlaceService placeService,
+            PathService pathService,
+            TimetableService timetableService,
+            SpeakerService speakerService,
+            LastUpdateService lastUpdateService,
+            ValidationErrorService validationErrorService) {
+        this.eventService = eventService;
+        this.placeService = placeService;
+        this.pathService = pathService;
+        this.timetableService = timetableService;
+        this.speakerService = speakerService;
+        this.lastUpdateService = lastUpdateService;
+        this.validationErrorService = validationErrorService;
+    }
+
+    @Transactional
+    @GetMapping(value = "/app-data")
+    public ResponseEntity<?> getAppData(){
+        AppData appData;
+        Set<String> roles = SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getAuthorities()
+                .stream()
+                .map(GrantedAuthority::getAuthority)
+                .collect(Collectors.toSet());
+        if (roles.contains("ROLE_USER")){
+            appData = new AppData(
+                    eventService.findActiveEvents(),
+                    placeService.findAllPlaces(),
```

Comment:
`findAllPlaces()` is ok? Every single other one contains "Active" keyword

Reply:
Yes, the majority of models are dependent of "active" located in path. I guess that only places are independent.